File: nodejs/test-fixtures/regressed/lib.test.js
import { test } from "node:test";
import assert from "node:assert";
import { add, multiply } from "./lib.js";

// REGRESSION test: assertion errada de proposito.
test("add failing on purpose", () => {
  assert.strictEqual(add(2, 3), 999);
});

test("multiply positive", () => {
  assert.strictEqual(multiply(3, 4), 12);
});

File: go/test-fixtures/regressed/lib_test.go
package qgregressed

import "testing"

// REGRESSION test: teste falhando de proposito.
func TestAddFailingOnPurpose(t *testing.T) {
	if Add(2, 3) != 999 {
		t.Fatalf("expected 999 (de proposito errado)")
	}
}

func TestMultiplyPositive(t *testing.T) {
	if Multiply(3, 4) != 12 {
		t.Fatalf("expected 12")
	}
}
